lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'license_detector.dart';
+
+@sealed
+class License {
+  /// SPDX identifier of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// Map from [Token.value] to their number of occurrences in this license.
+  final Map<String, int> occurrences;
+
+  License._(this.content, this.tokens, this.occurrences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, identifier);
+  }
+}
+
+String tokensNormalizedValue(Iterable<Token> tokens) {
+  return tokens.map((token) => token.value).join(' ');
+}
+
+/// Representation of n-gram consistent of `n` tokens.
+///
+/// The number of tokens `n` in the n-grams is determined by  confidence threshold for the match.
+///  See: [computeGranularity].
```

Comment:
nit: Extra whitespace at the beginning of the line.


```suggestion
/// See: [computeGranularity].
```